Melee and Box pickup detection fixed

mikegreber opened the pull request.

mikegreber committed:
Milk bottle tightened

mikegreber committed:
Melee and Box pickup hit detection rework

mikegreber committed:
Fixed box and melee hit detection

mikegreber committed:
Code cleanup

mikegreber committed:
Diaper bomb fullscreen fix

vppatel111 commented on HITDAD/js/plugins/SPFBoxMouseToss.js:
```diff
@@ -121,15 +111,58 @@
+        return pickupableEvents;
+    }
+
+    // Checks if and returns an enemy that is within melee range, returns null if none in range.
```
Comment is talking about enemies but this function is about boxes?

vppatel111 commented on HITDAD/js/plugins/SPFBoxMouseToss.js:
```diff
@@ -121,15 +111,58 @@
+        let direction = $gamePlayer.direction();
+
+        // So it scans from around the players back for detecting objects underneath
+        let xTraceStart = direction === 4 ? $gamePlayer._realX + 0.5 : $gamePlayer._realX  - 0.5;
```
`let xTraceStart = direction === DIRECTION.LEFT ? $gamePlayer._realX + 0.5 : $gamePlayer._realX  - 0.5;`

=> Which is defined in SPFCore.js

vppatel111 commented on HITDAD/js/plugins/SPFBoxMouseToss.js:
```diff
@@ -121,15 +111,58 @@
+            let verticalOffset = Math.abs($gamePlayer._realY - box._realY);
+
+            // Get box in direction player is facing to have positive distance value (direction === 6 means right)
+            let forwardDistanceToBox = direction === 6 ? distanceToBox : - distanceToBox;
```
In this case, you could simply use Math.abs(distanceToBox) to get the absolute value of distance to box.

mikegreber replied:
Math.abs does not work in this case as negative values are intended for objects behind the player, this function has been refactored to be easier to understand.

vppatel111 commented on HITDAD/js/plugins/SPFMeleeAttack.js:
```diff
@@ -38,102 +38,131 @@
+  //   return enemyEvent;
+  // }
+
+  // Checks if and returns an enemy that is within melee range.
```
Don't worry about commenting out code from now on, just delete it if your replacing it.

vppatel111 commented on HITDAD/js/plugins/SPFMeleeAttack.js:
```diff
@@ -38,102 +38,131 @@
+      let verticalOffset = Math.abs($gamePlayer._realY - enemy._realY);
+
+      // Get enemies in direction player is facing to have positive distance value (direction === 6 means right)
+      let forwardDistanceToEnemy = direction === 4 ?  distanceToEnemy : -distanceToEnemy;
```
Similarly Math.abs() here.

vppatel111 commented on HITDAD/js/plugins/SPFMeleeAttack.js:
```diff
@@ -38,102 +38,131 @@
+    //     break;
+    // }
+
+    // return []; // No collision with events.
```
Remove commented code entirely.

vppatel111 commented on HITDAD/js/plugins/SPFMeleeAttack.js:
```diff
@@ -38,102 +38,131 @@
-    }
 
-  }
+  // function PerformAttack() {
```
You get the gist be now.

mikegreber committed:
Box throw trajectory in progress

mikegreber committed:
Trajectory display prototype

Also SPF_LineTrace created for Box Pickip and Melee. Refactored to be easier to understand.

mikegreber committed:
Trajectory prototyped

mikegreber committed:
function position change

vppatel111 commented:
Tested and looks amazing!
